Auth: Add skip_org_role_sync setting to OAuth integration Google (#61572)
grafana/grafana: docs/sources/setup-grafana/configure-grafana/_index.md

The setting `oauth_skip_org_role_update_sync` will be deprecated in favor of provider-specific settings.

> The table below show the OAuth provider and their setting with the default value and the skip org role sync setting.

Reviewer: ```suggestion
The following table shows the OAuth provider's setting with the default value and the skip org role sync setting.
```

| OAuth Provider | `oauth_skip_org_role_sync_update` | `skip_org_role_sync` | Behavior |
| --- | --- | --- | --- |
| AzureAD | false | false | will sync with AzureAD roles |
| AzureAD | true | false | skip org role sync for OAuth providers including AzureAD users |
| AzureAD | false | true | skip org role sync for AzureAD users |
| AzureAD | true | true | skip org role sync for AzureAD users and all other OAuth providers |

### [auth.google] skip_org_role_sync

> Opon first login from a user we set the organization roles from the setting `AutoAssignOrgRole`. If you want to manage organizational roles, you can set the `skip_org_role_sync` option to `true`. Please note that there is also a separate setting called `oauth_skip_org_role_update_sync` which has a different scope. While `skip_org_role_sync` only applies to the specific OAuth provider, `oauth_skip_org_role_update_sync` is a generic setting that affects all configured OAuth providers.

Author: As we do not get roles from Google auth provider.

Here is the setting that we set eventually after the code goes through a user that logs in without a role assigned from the provider.
https://github.com/grafana/grafana/blob/main/pkg/services/user/userimpl/user.go#L151

> Opon first login from a user we set the organization roles from the setting `AutoAssignOrgRole`. If you want to manage organizational roles, you can set the `skip_org_role_sync` option to `true`. Please note that there is also a separate setting called `oauth_skip_org_role_update_sync` which has a different scope. While `skip_org_role_sync` only applies to the specific OAuth provider, `oauth_skip_org_role_update_sync` is a generic setting that affects all configured OAuth providers.

Reviewer: ```suggestion
Upon the first login from a user, we set the organization roles from the setting `AutoAssignOrgRole`. If you want to manage organizational roles, set the `skip_org_role_sync` option to `true`. 

> **Note:** There is a separate setting called `oauth_skip_org_role_update_sync` which has a different scope. While `skip_org_role_sync` only applies to the specific OAuth provider, `oauth_skip_org_role_update_sync` is a generic setting that affects all configured OAuth providers.
```

The setting `oauth_skip_org_role_update_sync` will be deprecated in favor of provider-specific settings.

> The table below show the OAuth provider and their setting with the default value and the skip org role sync setting.

Reviewer: ```suggestion
The following table shows the OAuth provider's setting with the default value and the skip org role sync setting.
```

| OAuth Provider | `oauth_skip_org_role_sync_update` | `skip_org_role_sync` | Behavior |
| --- | --- | --- | --- |

> | Google | false | false | user organization roles are set with defaultRole and will not be able to be changed |

Reviewer: ```suggestion
| Google | false | false | User organization roles are set with `defaultRole` and cannot be changed |
```

| Google | true | false | user organization roles are set with defaultRole for Google, for other providers the synchronization will be skipped and the org role will be able to be changed; along with other OAuth provider users org roles |

> | Google | false | true | user organization roles are set with defaultRole and the org role will be able to be changed for Google synced users |

Reviewer: ```suggestion
| Google | false | true | User organization roles are set with `defaultRole` and the org role can be changed for Google synced users. |
```

> | Google | true | true | user organization roles are set with defaultRole for Google, for other providers the synchronization will be skipped and the org role will be able to be changed; along with other OAuth provider users org roles |

Reviewer: ```suggestion
| Google | true | true | User organization roles are set with `defaultRole` for Google. For other providers, the synchronization will be skipped, and the org role can be changed, along with other OAuth provider users' org roles. |
```

The setting `oauth_skip_org_role_update_sync` will be deprecated in favor of provider-specific settings.

The following table shows the OAuth provider's setting with the default value and the skip org role sync setting.
| OAuth Provider | `oauth_skip_org_role_sync_update` | `skip_org_role_sync` | Behavior |
| --- | --- | --- | --- |
| AzureAD | false | false | will sync with AzureAD roles |
| AzureAD | true | false | skip org role sync for OAuth providers including AzureAD users |
| AzureAD | false | true | skip org role sync for AzureAD users |
| AzureAD | true | true | skip org role sync for AzureAD users and all other OAuth providers |

### [auth.google] skip_org_role_sync

Upon the first login from a user, we set the organization roles from the setting `AutoAssignOrgRole`. If you want to manage organizational roles, set the `skip_org_role_sync` option to `true`. 

> **Note:** There is a separate setting called `oauth_skip_org_role_update_sync` which has a different scope. While `skip_org_role_sync` only applies to the specific OAuth provider, `oauth_skip_org_role_update_sync` is a generic setting that affects all configured OAuth providers.

> The setting `oauth_skip_org_role_update_sync` will be deprecated in favor of provider-specific settings.

Reviewer 2: nit: should this be a part of the above note?

Author: yea, it is a bit verbose currently. I want to review the docs at some point soon, to either make it more concise or to put all the tables together of sorts.

I will remove this line
